pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -286,31 +286,46 @@ func (k *KSMCheck) Configure(senderManager sender.SenderManager, integrationConf
 	builder := kubestatemetrics.New()
 	builder.WithUsingAPIServerCache(k.instance.UseAPIServerCache)
 
-	// Due to how init is done, we cannot use GetAPIClient in `Run()` method
-	// So we are waiting for a reasonable amount of time here in case.
-	// We cannot wait forever as there's no way to be notified of shutdown
-	apiCtx, apiCancel := context.WithTimeout(context.Background(), maximumWaitForAPIServer)
-	defer apiCancel()
-	c, err := apiserver.WaitForAPIClient(apiCtx)
-	if err != nil {
-		return err
-	}
+	k.configurePodCollection(builder, k.instance.Collectors)
 
-	// Discover resources that are currently available
-	resources, err := discoverResources(c.Cl.Discovery())
-	if err != nil {
-		return err
-	}
+	var collectors []string
+	var apiServerClient *apiserver.APIClient
+	var resources []*v1.APIResourceList
 
-	// Prepare the collectors for the resources specified in the configuration file.
-	collectors, err := filterUnknownCollectors(k.instance.Collectors, resources)
-	if err != nil {
-		return err
-	}
+	switch k.instance.PodCollectionMode {
+	case nodeKubeletPodCollection:
+		// In this case we don't need to set up anything related to the API
+		// server.
+		collectors = []string{"pods"}
```

Comment:
Wouldn't it be possible to have a check configuration with a set of collectors on top of the pod collection mode? In that case, should we fail or log a warning at least?

Reply:
That is not supported. I added a comment in the code to explain why: https://github.com/DataDog/datadog-agent/blob/359ea396b90b2f9a798c2949f6204e8560bdd3a0/pkg/collector/corechecks/cluster/ksm/kubernetes_state.go#L862

And there's a warning a few lines below that one: https://github.com/DataDog/datadog-agent/blob/359ea396b90b2f9a798c2949f6204e8560bdd3a0/pkg/collector/corechecks/cluster/ksm/kubernetes_state.go#L869